test/NuGet.Core.Tests/NuGet.Common.Test/RuntimeEnvironmentHelperTests.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace NuGet.Common.Test
+{
+    public class RuntimeEnvironmentHelperTests
+    {
+        [Fact]
+        public void RuntimeEnvironmentHelper_MutuallyExclusiveRuntimes()
+        {
+            // It only should return true in either Windows or MacOS/Linux
+            Assert.True(RuntimeEnvironmentHelper.IsWindows ^ (RuntimeEnvironmentHelper.IsMacOSX | RuntimeEnvironmentHelper.IsLinux));
```

Comment:
You're trying to be too clever with logical operators, and yet there's a bug here that belies the claim (or spirit) of your test name.

If `IsWindows` returns `false` and both `IsMacOSX` and `IsLinux` return `true` this test will pass.

This is a more clear and correct test:
```
var results = new[] { RuntimeEnvironmentHelper.IsWindows, RuntimeEnvironmentHelper.IsMacOSX, RuntimeEnvironmentHelper.IsLinux };

Assert.Equal(1, results.Count(result => result));
```